server/src/server.ts

```diff
@@ -849,6 +877,15 @@ function onMessage(msg: p.Message) {
       };
       send(response);
     } else if (msg.method === "initialize") {
+      // Save initial configuration, if present
+      let initParams = msg.params as InitializeParams;
+      let initialConfiguration = initParams.initializationOptions
+        ?.extensionConfiguration as extensionConfiguration | undefined;
+
+      if (initialConfiguration != null) {
+        extensionConfiguration = initialConfiguration;
```

Comment:
What happens without this change? What behaviour is affected?
Asking as this is not just about the new configuration option just introduced.